docs/defaulting.md

```diff
@@ -0,0 +1,180 @@
+  <name>testvm</name>
+  <uuid>a8335812-53d4-4e57-ac0b-72951c885a57</uuid>
+  <memory unit='KiB'>62500</memory>
+  <currentMemory unit='KiB'>62500</currentMemory>
+  <vcpu placement='static'>1</vcpu>
+  <os>
+    <type arch='x86_64' machine='pc-i440fx-2.7'>hvm</type>
+    <boot dev='hd'/>
+  </os>
+  <clock offset='utc'/>
+  <on_poweroff>destroy</on_poweroff>
+  <on_reboot>restart</on_reboot>
+  <on_crash>destroy</on_crash>
+  <devices>
+    <emulator>/usr/bin/qemu-system-x86_64</emulator>
+    <controller type='usb' index='0' model='piix3-uhci'>
+      <address type='pci' domain='0x0000' bus='0x00' slot='0x01' function='0x2'/>
+    </controller>
+    <controller type='pci' index='0' model='pci-root'/>
+    <interface type='network'>
+      <mac address='52:54:00:2e:84:30'/>
+      <source network='default'/>
+      <model type='rtl8139'/>
+      <address type='pci' domain='0x0000' bus='0x00' slot='0x02' function='0x0'/>
+    </interface>
+    <input type='mouse' bus='ps2'/>
+    <input type='keyboard' bus='ps2'/>
+    <memballoon model='virtio'>
+      <address type='pci' domain='0x0000' bus='0x00' slot='0x03' function='0x0'/>
+    </memballoon>
+  </devices>
+</domain>
+```
+
+Finally, when a Domain was successfully started by Libvirt, Libvirt assigns an
+`alias` to every device in the XML.
+
+## The Problem
+
+ * Hotplug operations are not possible with the minimal VM spec of kubevirt,
+   since we could match wrong devices.
+ * After VM restarts, different devices or addresses could be assigned to a VM
+   by libvirt
+ * Libvirt assigns all that data on the host, either when defining the Domain,
+   or after the Domain was started. That requires asynchronous lookups, and
+   when thinking about administrating offline VMs, it binds your domain
+   specification management to the kubevirt runtime.
+
+If your VMs are ephemeral (no migrations, recreating VMs vs. restarting VMs, no
+hotplug), you are not affected by that.
+
+## Proposed Solution
+
+### Defaulting service based on libvirt
+
+Except for some runtime information, which is normally not included in
+migratable domain XMLs, libvirt fills in all defaults when defining a domain.
+We can leverage that with a simple REST based service, which defines VMs in an
+isolated libvirt (running libvirt and the small REST service in a Pod). The
+service is a standalone component in KubeVirt and can then be integrated into
+two different flows:
+
+ 1. When doing a POST to
+    `/apis/kubevirt.io/v1alpha1/namespaces/mynamespace/vms`, `virt-api` can do
+    a roundtrip to that service and prefill the defaults. As the response of
+    the POST to `virt-api`, you will get the completely prefilled VM
+    specification. This updated spec contains all the necessary bits, to allow
+    consistent hotplugging and keeping the VM definition consistent between
+    redefines.
+ 2. Add an extra endpoint to `virt-api` which only purpose is to prefill the VM
```

Comment:
I want to make sure I understand the two options here.

Option 1. automatically prefills in defaults when a VMConfig is posted for the first time.

Option 2. Provides an endpoint for users to post a VMConfig to and receive the populated VMConfig back. 

If my understanding is correct, with option 1 we attempt to solve the issue within kubevirt automatically, With option 2 we push this logic to the administrator and provide an endpoint to help them with populating the VM spec.

If we can abstract away this complexity from the administrator, that is my preference (option 1).  What is the concern about hotplugging with option 1? Understanding the flow someone would go through to hotplug a device might make this more obvious. It is unclear to me what object (VMConfig or VM) someone is modifying to hotplug.